### ISSUE TITLE
Heading order skips from h1 to h3 on Service Landing Page

### ISSUE THREAD

**Issue by keelanfh** [localgovdrupal/localgov_services#188]
Apologies if wrong repo for this...

e.g. on https://demo.localgovdrupal.org/births-deaths-marriages-and-citizenship

The page title ("Births, deaths, marriages and citizenship") is a `h1`, but the links to child pages (e.g. "Registering a birth") are `h3`. This means there is an illogical heading order, which could be an accessibility issue (https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships). 

It is not possible to change this in the CMS, as it would be for most headings, as the child pages are entered in the form.

**Comment by andybroomfield** [comment]
Just to add this issue also came up in BHCC accessbility report.
It happens because the card template we use in our custom theme uses a h3, I assume the teaser template in localgov_base is the same.
It was set as h3 as on the sub landing page, the same template can also be used, however there are h2 on the sublanding pages for all the topics that group the links, which is why I assume they are h3s.

Our current fix PR is to remove the h3s and just use divs. Not ideal though as they are links that might be acceptable.

**Comment by keelanfh** [comment]
Thanks for that @andybroomfield, useful to know that it's cropping up for other councils too.

You're right that the h3 is added in `localgov_base` and not here. @markconroy maybe we should move this issue over to `localgov_base`, though I'm not quite sure how the two interact.

The issue can crop up on service sub landing pages too, if people don't use the h2s, which are optional.

The actual change is of course really small - https://github.com/essexcountycouncil/localgov_base/commit/1b3961785c3976026a1d41120f121c5e37d7d2b3 - but when I've added this, it then introduces an illogical heading order for anyone using headings on service sub-landing pages.

Not sure if it's possible to have something that's 'aware' of the structure of the rest of the page and set the right heading level accordingly?

**Comment by andybroomfield** [comment]
I've not found a way to pass through anything that can be context aware, as the template is a node template (should be possible with a field template). Ideally we would like a way of passing in a heading tag value, calculated on if there is a topic heading section (or this is a landing page). For the moment our plan is to remove header tags altogether.

**Comment by markconroy** [comment]
This is a _really_ tricky issue to solve, and one that comes up in CMSs all the time. If you look at the default node.html.twig template in Drupal, those titles are set to `h2` which is what I usually leave them at.

In LGD they were set to H3 before I came along, which was correct in many of our places as we have a page title block using `h1`, then a section title using `h2` and then our node/card/teaser title using `h3`. But, when the same template is used (correctly) in a different context the heading order is wrong. 

I'm not sure it's fixable in localgov_base for all sites in all situations. I think you might need to change the template per site to have a clause in it that checks for a parent and if it has a parent, then it uses `h3` and if not it uses `h2`. I've done that on some annertech sites, if the node is rendered via an entity reference field, because you can check the `referringItem` and see what is in that.

It's definitely on my "to think about" list.

**Comment by keelanfh** [comment]
Thanks @markconroy, that's a really helpful explanation.

Another option I thought about was to set the heading level as an option in the CMS - I think we do this somewhere else? Although that would need to be set on every page and so is prone to errors.

This is easy enough for us to fix on essex.gov.uk for now, since we're not planning on using service sub-landing pages at this point and so we can just set `h2` everywhere. But it's great to hear that it's something you're thinking about!

**Comment by markconroy** [comment]
You can see an example of what I mentioned about in the `localgov_base` theme for the box links heading.

https://github.com/localgovdrupal/localgov_base/blob/1.x/templates/paragraphs/paragraph--localgov-box-link.html.twig#L48

**Comment by keelanfh** [comment]
Was chatting to @MatthewWilkes earlier about this and he suggested another approach:

Include a `visually-hidden` `h2` above the links where there isn't a `h2`. [This blog post](https://www.nomensa.com/blog/how-structure-headings-web-accessibility) mentions where this was done on the gov.uk homepage. 

This might be easier to implement?

**Comment by markconroy** [comment]
A visually hidden h2 above the links is fine, but only if it's got content in it that is relevant. We don't want, for example, an empty h2, just so we can pass automated a11y checkers.

If you look at the example in the blog post from Gov.uk, they have a section heading in the h2 and then render the sub items as h3, which is semantically correct.

Good content design would dictate that the H2 in this case for a heading _would_ be used, though maybe we could add in a boolean for _hide this heading from all except screenreaders_.

**Comment by finnlewis** [comment]
I couldn't see any h2 on this service sub landing page, so I edited it and added "Another heading" - I see now:  https://demo.localgovdrupal.org/libraries-leisure-and-culture/z-sports-and-activities

![image](https://user-images.githubusercontent.com/326588/215750572-d1e9c15e-4d08-43f7-b9a0-1733f2aacaa9.png)

Can we add some template logic that calls an appropriate level teaser card or passes the current heading level into the template for conditional headings?

**Comment by keelanfh** [comment]
Does anyone want to add a h2 to this module for this, for people not using the base theme? I'm not really familiar enough with this to know where to put it!

**Comment by markconroy** [comment]
@keelanfh it just needs to be added to this template https://github.com/localgovdrupal/localgov_services/blob/2.x/modules/localgov_services_landing/templates/node--localgov-services-landing--full.html.twig
